Doc/library/tty.rst

```diff
@@ -1,41 +1,98 @@
-:mod:`tty` --- Terminal control functions
-=========================================
+:mod:`tty` --- Termios module extensions
+========================================
 
 .. module:: tty
    :platform: Unix
-   :synopsis: Utility functions that perform common terminal control operations.
+   :synopsis: Termios module extensions
 
 .. moduleauthor:: Steen Lumholt
 .. sectionauthor:: Moshe Zadka <moshez@zadka.site.co.il>
+.. sectionauthor:: Soumendra Ganguly <soumendra@tamu.edu>
 
 **Source code:** :source:`Lib/tty.py`
 
 --------------
 
-The :mod:`tty` module defines functions for putting the tty into cbreak and raw
-modes.
+The :mod:`tty` module extends the termios module by defining utility functions
+and classes.
 
 Because it requires the :mod:`termios` module, it will work only on Unix.
 
-The :mod:`tty` module defines the following functions:
+winsize Objects
+---------------
+
+.. data:: HAVE_WINSZ
+
+   :const:`True` if both :const:`termios.TIOCGWINSZ` and :const:`termios.TIOCSWINSZ`
+   are defined; :const:`False` otherwise. The :class:`winsize` objects are implemented
+   if and only if this is :const:`True`.
+
+
+.. class:: winsize(row=0, col=0, xpixel=0, ypixel=0, fd=None)
+
+   Create a :class:`winsize` object, which represents a struct winsize; the
+   attributes are *ws_row*, *ws_col*, *ws_xpixel*, and *ws_ypixel*; these are
+   all nonnegative integers. If *fd* is not :const:`None`, then get window
+   size of tty of which *fd* is a file descriptor. If *fd* is :const:`None`,
+   then set the attributes based on the values of *row*, *col*, *xpixel*, and
+   *ypixel*.
+
+
+.. method:: winsize.tcgetwinsize(fd)
+
```

Comment:
Naming justification: they have decided to include functions tcgetwinsize() and tcsetwinsize() in IEEE Std 1003.1 ("POSIX.1") issue 8 [ the upcoming version of POSIX ]; see https://www.austingroupbugs.net/view.php?id=1151#c3856.